src/psyclone/psyir/nodes/intrinsic_call.py

```diff
@@ -1593,7 +2190,11 @@ class Intrinsic(IAttr, Enum):
                 types=DataNode,
                 arg_names=(("i", "j"),)),
             optional_args={},
-            return_type=None,
+            return_type=lambda node: (
+                node.argument_by_name("i").datatype.copy()
```

Comment:
I think you may have this same code 3 times now so worth a helper function instead?

Reply:
Modified to use existing functions.